### ISSUE TITLE
use metadata type, not dict, in main

### ISSUE THREAD

**Issue by williballenthin** [mandiant/capa#1411]
while this is what's required to make this work today, i don't like how it encourages us to keep using untyped Dicts to represent our intermediate results. i wonder if we should update the surrounding code (such as whats used in `main.py`) to use the `result_document.Metadata` objects instead.

_Originally posted by @williballenthin in https://github.com/mandiant/capa/pull/1396#discussion_r1149171596_

**Comment by manasghandat** [comment]
@williballenthin Can you assign me this issue. Would like to work on it.

**Comment by Aayush-Goel-04** [comment]
@williballenthin , Regarding updating meta of type(Metadata) I think it would be better to create `from_capa` apis for other classes like Layout, Feature_Counts in `result_document.py`. Since while updating meta from dictionary causes type error between dict and tuple.

**Comment by Aayush-Goel-04** [comment]
> we should maybe just return `result_document.Metadata` here and use it throughout the code.

If we are using meta as type Metadata then https://github.com/mandiant/capa/blob/af1500825af9454e716b05548360d45393c87295/capa/main.py#LL1232C5-L1239C69
Here meta is passed to `ResultDocument.from_capa(meta)` in `capa.render.json/verbose/vverbose/default.render` where it is processed as `Metadata.from_capa(meta)` thus meta should be of type dict. I guess some changes will also be required in render too.

**Comment by williballenthin** [comment]
> I think it would be better to create from_capa apis for other classes like Layout, Feature_Counts in result_document.py

Yup, that sounds great.

**Comment by williballenthin** [comment]
> I guess some changes will also be required in render too.

yes, i think thats right. we should prefer to use the `Metadata` type wherever possible and avoid the dict representation (unless actively serializing to a file).

**Comment by mr-tz** [comment]
closed by #1502, thanks @Aayush-Goel-04!
